test/refactoring/ConvertAsync/convert_bool.swift

```diff
@@ -0,0 +1,382 @@
+optionalBoolWithErr { str, optBool, b, err in
+  if optBool != nil {
+    print("a \(err!)")
+  }
+  if optBool == nil {
+    print("b \(err!)")
+  }
+  if optBool == true {
+    print("c \(err!)")
+  }
+  if ((optBool) == (false)) {
+    print("d \(err!)")
+  }
+  if optBool == false {
+    print("e \(err)")
+  }
+  if optBool != true {
+    print("f \(err!)")
+  }
+  if b {
+    print("g \(err!)")
+  }
+}
+
+// It's a little unfortunate that print("a \(<#err#>!)") gets classified in the success
+// block below, but it doesn't seem like a case that's likely to come up, as optBool
+// would then be inaccessible in the error block.
+
+// OPT-BOOL-WITH-ERR:      do {
+// OPT-BOOL-WITH-ERR-NEXT:   let (str, optBool, b) = try await optionalBoolWithErr()
+// OPT-BOOL-WITH-ERR-NEXT:   print("a \(<#err#>!)")
+// OPT-BOOL-WITH-ERR-NEXT:   if <#optBool#> == false {
+// OPT-BOOL-WITH-ERR-NEXT:     print("e \(<#err#>)")
+// OPT-BOOL-WITH-ERR-NEXT:   }
+// OPT-BOOL-WITH-ERR-NEXT:   if <#optBool#> != true {
+// OPT-BOOL-WITH-ERR-NEXT:     print("f \(<#err#>!)")
+// OPT-BOOL-WITH-ERR-NEXT:   }
+// OPT-BOOL-WITH-ERR-NEXT: } catch let err {
+// OPT-BOOL-WITH-ERR-NEXT:   print("b \(err)")
+// OPT-BOOL-WITH-ERR-NEXT:   print("c \(err)")
+// OPT-BOOL-WITH-ERR-NEXT:   print("d \(err)")
+// OPT-BOOL-WITH-ERR-NEXT:   print("g \(err)")
+// OPT-BOOL-WITH-ERR-NEXT: }
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path | %FileCheck -check-prefix=OBJC-BOOL-WITH-ERR %s
+ClassWithHandlerMethods.firstBoolFlagSuccess("") { str, success, unrelated, err in
+  if !unrelated {
+    print(err!)
+  }
+  if !success {
+    print("oh no")
+  }
+  if !success {
+    print(err!)
+  }
+  if success {
+    print("woo")
+  }
+  if str != nil {
+    print("also woo")
+  }
+}
+
+// OBJC-BOOL-WITH-ERR:      do {
+// OBJC-BOOL-WITH-ERR-NEXT:   let (str, success, unrelated) = try await ClassWithHandlerMethods.firstBoolFlagSuccess("")
+// OBJC-BOOL-WITH-ERR-NEXT:   if !unrelated {
+// OBJC-BOOL-WITH-ERR-NEXT:     print(<#err#>!)
+// OBJC-BOOL-WITH-ERR-NEXT:   }
+// OBJC-BOOL-WITH-ERR-NEXT:   print("woo")
+// OBJC-BOOL-WITH-ERR-NEXT:   print("also woo")
+// OBJC-BOOL-WITH-ERR-NEXT: } catch let err {
+// OBJC-BOOL-WITH-ERR-NEXT:   print("oh no")
+// OBJC-BOOL-WITH-ERR-NEXT:   print(err)
+// OBJC-BOOL-WITH-ERR-NEXT: }
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path | %FileCheck -check-prefix=OBJC-BOOL-WITH-ERR2 %s
+ClassWithHandlerMethods.secondBoolFlagFailure("") { str, unrelated, failure, err in
+  if unrelated {
+    print(err!)
+  }
+  if failure {
+    print("oh no")
+  }
+  if failure {
+    print(err!)
+  }
+  if !failure {
+    print("woo")
+  }
+  if str != nil {
+    print("also woo")
+  }
+  if failure && err == nil {
+    print("wat")
+  }
+  if failure && err != nil {
```

Comment:
Do we handle `if failure, let err = err`?

Reply:
Yes, let me add a test for that along with an `else if` test

Reply:
Done, how does this look?